### PR TITLE
Verify resource exists before sync

### PR DESCRIPTION
**Pull request by npolshakova**
When syncing the output snapshot to local cluster storage only write resources that exist. 
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/987

### CODE REVIEW

File: contrib/pkg/output/snapshot.go [hunk 1]
```diff
@@ -207,53 +208,91 @@ type Snapshot struct {
 	ListsToSync []ResourceList
 }
 
+// Options for writing resources of a given type
+type OutputOpts struct {
```
Comment (ashleywang1) [inline contrib/pkg/output/snapshot.go:212]: nice!

File: contrib/pkg/output/snapshot.go [hunk 2]
```diff
@@ -207,53 +208,91 @@ type Snapshot struct {
...
-					errHandler.HandleWriteError(object, err)
+					opts.ErrHandler.HandleWriteError(object, err)
 				}
 				continue
 			}
 
+			for _, resource := range listForCluster {
+				if opts.Verifier != nil {
+					gvk := resource.GetObjectKind().GroupVersionKind()
+					resourceRegistered, _ := opts.Verifier.VerifyServerResource(
```
Comment (ashleywang1) [inline contrib/pkg/output/snapshot.go:278]: Can we either debug log the error here, or add a comment on why it's ok to ignore it?

File: contrib/pkg/output/snapshot.go [hunk 3]
```diff
@@ -207,53 +208,91 @@ type Snapshot struct {
...
+func (s Snapshot) SyncLocalCluster(ctx context.Context, cli client.Client, opts OutputOpts) {
 
+	var presentResources []ezkube.Object
 	for _, list := range s.ListsToSync {
 		listForLocalCluster := list.SplitByClusterName()[multicluster.LocalCluster]
 
+		for _, resource := range listForLocalCluster {
+			if opts.Verifier != nil {
+				gvk := resource.GetObjectKind().GroupVersionKind()
+				resourceRegistered, _ := opts.Verifier.VerifyServerResource(
```
Comment (ilackarms) [inline contrib/pkg/output/snapshot.go:232]: we should check the error and report it to the err handler (and `continue`)

File: contrib/pkg/output/snapshot.go [hunk 4]
```diff
@@ -207,53 +208,91 @@ type Snapshot struct {
...
 // only writes resources intended for the local cluster (with ClusterName == "")
 // Note that Syncing snapshots in this way adds the label
-func (s Snapshot) SyncLocalCluster(ctx context.Context, cli client.Client, errHandler ErrorHandler) {
+func (s Snapshot) SyncLocalCluster(ctx context.Context, cli client.Client, opts OutputOpts) {
 
+	var presentResources []ezkube.Object
 	for _, list := range s.ListsToSync {
 		listForLocalCluster := list.SplitByClusterName()[multicluster.LocalCluster]
 
+		for _, resource := range listForLocalCluster {
```
Comment (ilackarms) [inline contrib/pkg/output/snapshot.go:229]: i don't think we need to iterate through each resource in each list; just check the gvk for each list

File: contrib/pkg/output/snapshot.go [hunk 5]
```diff
@@ -207,53 +208,91 @@ type Snapshot struct {
...
+			if opts.Verifier != nil {
+				gvk := resource.GetObjectKind().GroupVersionKind()
```
Comment (ilackarms) [inline contrib/pkg/output/snapshot.go:231]: we don't set these when we translate out objects, we may need to use `apiutil.GVKForObject` here. another option is to just pass the gvks in from generated code

**Review by ilackarms — APPROVED** [review]

**Pull request merged**